tests/interpolate_test.py

```diff
@@ -53,3 +57,22 @@ def run_around_tests(self):
         yield
         # Check there are no side effects
         assert np.all(t == self.t)
+
+
+class TestRawTrajectories:
+    def setup_method(self):
+        tr = np.load(cons.test_raw_trajectories_path, allow_pickle=True)
+        interpolate_nans(tr)
+        self.tr = tr
+
+    @pytest.mark.parametrize(
+        "frame_fraction,smooth_sigma",
+        [(random.random(), random.random()) for _ in range(5)],
+    )
+    def test_consistent(self, frame_fraction, smooth_sigma):
+        smooth_tr = smooth(self.tr, sigma=smooth_sigma, only_past=True)
+        s, v, a = velocity_acceleration_backwards(smooth_tr)
+        frame = int((len(s) - 2) * frame_fraction)
+        print(f"Testing in frame {frame} with smooth sigma {smooth_sigma}")
+        np.testing.assert_allclose(s[frame] + v[frame + 1], s[frame + 1])
```

Comment:
I would have expected `s[frame] + v[frame] = s[frame+1]`. I imagine it is not the case because of the `only_past=True`, right? or am I missing something else?

Reply:
exactly, it is because only_past=True

It is important to generate trajectories because the action of the agent is `a[t+1]`. Using that action and current velocity, we calculate future velocity. Using future velocity and current location, we calculate future location.